packages/mdc-switch/mdc-switch.scss

```diff
@@ -45,79 +38,86 @@
-      position: absolute;
-      top: 0;
-      right: 0;
-      bottom: 0;
-      left: 0;
-      transition:
-        mdc-switch-transition(opacity),
-        mdc-switch-transition(background-color);
-      border-radius: 7px;
-      opacity: .38;
-      content: "";
-    }
+// postcss-bem-linter: ignore
+.mdc-switch__knob {
+  @include mdc-elevation(2);
+
+  display: block;
+  position: absolute;
+  top: $mdc-switch-knob-vertical-offset_;
+  left: 0;
+  width: $mdc-switch-knob-diameter;
+  height: $mdc-switch-knob-diameter;
+  transform: translateX(0);
+  transition:
+    mdc-switch-transition(transform),
+    mdc-switch-transition(background-color);
+  border-radius: $mdc-switch-knob-diameter / 2;
+  z-index: 1;
+
+  // Focus indicator
+  // postcss-bem-linter: ignore
+  &::before {
+    @include mdc-switch-tap-target_;
 
-    // postcss-bem-linter: ignore
-    .mdc-switch__knob {
-      @include mdc-elevation(2);
-
-      display: block;
-      position: absolute;
-      top: $mdc-switch-knob-vertical-offset_;
-      left: 0;
-      width: $mdc-switch-knob-diameter;
-      height: $mdc-switch-knob-diameter;
-      transform: translateX(0);
-      transition:
-        mdc-switch-transition(transform),
-        mdc-switch-transition(background-color);
-      border-radius: $mdc-switch-knob-diameter / 2;
-      z-index: 1;
-
-      // Focus indicator
-      // postcss-bem-linter: ignore
-      &::before {
-        @include mdc-switch-tap-target_;
-
-        transform: scale(0);
-        transition:
-          mdc-switch-transition(transform),
-          mdc-switch-transition(background-color);
-        border-radius: $mdc-switch-focus-ring-diameter / 2;
-        opacity: .2;
-        content: "";
-      }
-    }
+    transform: scale(0);
+    transition:
+      mdc-switch-transition(transform),
+      mdc-switch-transition(background-color);
+    border-radius: $mdc-switch-focus-ring-diameter / 2;
+    opacity: .2;
+    content: "";
   }
 }
 
+@at-root {
+  @include mdc-switch-unchecked-track-color_($mdc-switch-unchecked-track-color);
+  @include mdc-switch-unchecked-knob-color_($mdc-switch-unchecked-knob-color);
+  @include mdc-switch-unchecked-focus-indicator-color_($mdc-switch-unchecked-focus-ring-color);
+  @include mdc-switch-track-color($mdc-switch-baseline-theme-color);
+  @include mdc-switch-knob-color($mdc-switch-baseline-theme-color);
+  @include mdc-switch-focus-indicator-color($mdc-switch-baseline-theme-color);
+}
+
 .mdc-switch__native-control:focus ~ .mdc-switch__background {
-  .mdc-switch__knob {
-    // Focus indicator
-    &::before {
-      position: absolute;
-      width: $mdc-switch-focus-ring-diameter;
-      height: $mdc-switch-focus-ring-diameter;
-      transform: scale(1);
-      transition:
-        mdc-switch-transition(transform),
-        mdc-switch-transition(background-color);
-      border-radius: $mdc-switch-focus-ring-diameter / 2;
-    }
+  // Focus indicator
+  .mdc-switch__knob::before {
```

Comment:
AFAICT, most of the properties below are redundant. They're already being set in
`.mdc-switch__knob::before` (or `mdc-switch-tap-target_`) above.

Keep `transform` and remove the rest (but make sure it doesn't break anything 😄 )

Reply:
Good catch...I didn't pay attention to the body of this rule. I think all I did was change the indentation.

I also noticed a few more duplications which I'm removing.